### DISCUSSION TITLE
Error when deploying my App

### DISCUSSION THREAD

**Discussion by TQuantab** [sveltejs/kit#12041] [Q&A] [answered]
Hi =)
im currently having a problem, trying to build my app for vercel.

Whenever building, im getting this error:
```js
error during build:
RollupError: src/routes/+layout.ts (1:35): "PUBLIC_SUPABASE_URL" is not exported by "virtual:$env/static/public", imported by "src/routes/+layout.ts".
```

The referenced part of the file looks like this:
```js
import { PUBLIC_SUPABASE_ANON_KEY, PUBLIC_SUPABASE_URL } from '$env/static/public';
import { createSupabaseLoadClient } from '@supabase/auth-helpers-sveltekit';
import { getAvatarUrl } from '$lib/utils';

export const load = async ({ fetch, data, depends }) => {
	depends('supabase:auth');

	const supabase = createSupabaseLoadClient({
		supabaseUrl: PUBLIC_SUPABASE_URL,
		supabaseKey: PUBLIC_SUPABASE_ANON_KEY,
		event: { fetch },
		serverSession: data.session
	});
```

Does anyone have an idea? Im pretty stuck right now, and already tried a ton to fix it.

**Comment by ra-kunnn** [comment]
i have the same problem i still dont know why, my .env.local looks fine

**Reply by joshuatreepc** [reply]
Did you fix it?

**Reply by TQuantab** [reply, marked as answer]
I actually managed to fix it. I don't remember exactly what I did, but I think you have to add the env inside vercel manually.

**Reply by ra-kunnn** [reply]
yes i was able to fix it, the exact same as what op did haha. i think theres an integration with supabase in the settings that does it automatically however if u use smth different in ur code to refer to ur public keys, u have to add them in your env inside vercel

**Comment by seb-goodman** [comment]
+1, but with GCP Cloud Run. Env variables also set within GCP.
